src/elements/Table/TableContext.tsx

```diff
@@ -0,0 +1,38 @@
+import { TableNode, TableCellNode } from '@prezly/story-content-format';
+import React, { useContext, createContext, PropsWithChildren } from 'react';
+
+interface ContextProps {
+    isHeaderCell: (cell: TableCellNode) => boolean;
+}
+
+const TableContext = createContext<ContextProps | null>(null);
+
+interface Props {
+    table: TableNode;
+}
+
+export function TableContextProvider({ table, children }: PropsWithChildren<Props>) {
+    const isHeaderCell = React.useCallback(
+        (cell: TableCellNode) => {
+            return Boolean(
+                (table.header?.includes(TableNode.TableHeader.FIRST_ROW) && isFirstRow(table, cell)) ||
+                    (table.header?.includes(TableNode.TableHeader.FIRST_COLUMN) && isFirstColumn(table, cell)),
+            );
+        },
+        [table],
+    );
+
+    return <TableContext.Provider value={{ isHeaderCell }}>{children}</TableContext.Provider>;
+}
+
+export function useTableContext() {
+    return useContext(TableContext);
+}
```

Comment:
What do you think if instead of `useTableContext()` we export `useIsHeader()`? 

```ts
export function useIsHeader(cell: TableCellNode) {
   const ctx = useTableContext();
   return ctx?.isHeaderCell(cell) ?? false;
}
```

Reply:
in that case we will not be able to use this hook inside another function. i think if there will be more stuff related table header we could extract it to separate hook